unyt/_array_functions.py

```diff
@@ -158,38 +158,160 @@ def _sanitize_range(_range, units):
     return new_range.squeeze()
 
 
-@implements(np.histogram)
-def histogram(
-    a,
-    bins=10,
-    range=None,
-    *args,
-    **kwargs,
-):
-    range = _sanitize_range(range, units=[a.units])
-    counts, bins = np.histogram._implementation(
-        np.asarray(a), bins, range, *args, **kwargs
-    )
-    return counts, bins * a.units
-
-
-@implements(np.histogram2d)
-def histogram2d(x, y, bins=10, range=None, *args, **kwargs):
-    range = _sanitize_range(range, units=[x.units, y.units])
-    counts, xbins, ybins = np.histogram2d._implementation(
-        np.asarray(x), np.asarray(y), bins, range, *args, **kwargs
-    )
-    return counts, xbins * x.units, ybins * y.units
-
-
-@implements(np.histogramdd)
-def histogramdd(sample, bins=10, range=None, *args, **kwargs):
-    units = [_.units for _ in sample]
-    range = _sanitize_range(range, units=units)
-    counts, bins = np.histogramdd._implementation(
-        [np.asarray(_) for _ in sample], bins, range, *args, **kwargs
-    )
-    return counts, tuple(_bin * u for _bin, u in zip(bins, units))
+def _histogram(a, bins=10, range=None, density=None, weights=None, normed=None):
```

Comment:
Since this is a private function, let's make the signature robust against dev mistakes
```suggestion
def _histogram(a, *, bins=10, range=None, density=None, weights=None, normed=None):
```